src/lib/forms/AccordionField.js

```diff
@@ -1,57 +1,86 @@
 // This file is part of React-Invenio-Forms
 // Copyright (C) 2020 CERN.
 // Copyright (C) 2020 Northwestern University.
-//
 // React-Invenio-Forms is free software; you can redistribute it and/or modify it
 // under the terms of the MIT License; see LICENSE file for more details.
 
 import React, { Component, useState } from "react";
 import PropTypes from "prop-types";
 import { Field, FastField } from "formik";
-import { Accordion, Container, Icon } from "semantic-ui-react";
+import { Accordion, Container, Icon, Label } from "semantic-ui-react";
 import _omit from "lodash/omit";
-import _get from "lodash/get";
+import { flattenAndCategorizeErrors } from "../utils";
 
 export class AccordionField extends Component {
-  hasError(errors, initialValues = undefined, values = undefined) {
-    const { includesPaths } = this.props;
-    for (const errorPath in errors) {
-      for (const subPath in errors[errorPath]) {
-        const path = `${errorPath}.${subPath}`;
-        if (
-          _get(initialValues, path, "") === _get(values, path, "") &&
-          includesPaths.includes(`${errorPath}.${subPath}`)
-        )
-          return true;
+  // Checks if there are any errors that match the given paths.
+
+  hasError(errors, includesPaths) {
+    return Object.keys(errors).some((errorPath) =>
+      includesPaths.some((path) => errorPath.startsWith(path))
+    );
+  }
+
+  // Generates a summary of errors categorized by severity.
+  getErrorSummary = (errors, includePaths, severityChecks) => {
+    const count = {};
+
+    // Count generic errors
+    for (const path in errors.flattenedErrors) {
+      if (includePaths.some((includePath) => path.startsWith(includePath))) {
+        count["error"] = (count["error"] || 0) + 1;
       }
     }
-    return false;
-  }
+
+    // Count severity-based errors
+    for (const key in errors.severityChecks) {
+      const severity = errors.severityChecks[key]?.severity;
+      const path = key;
+
+      if (
+        severity &&
+        includePaths.some((includePath) => path.startsWith(includePath))
+      ) {
+        count[severity] = (count[severity] || 0) + 1;
+      }
+    }
+
+    // Format output to display labels
+    // e.g., { error: "1 Error", warning: "2 Warnings" }
+    const formattedCount = {};
+    for (const [severity, num] of Object.entries(count)) {
+      const label =
+        severityChecks?.[severity]?.label ||
+        severity.charAt(0).toUpperCase() + severity.slice(1);
+      formattedCount[severity] = `${num} ${label}${num === 1 ? "" : "s"}`;
```

Comment:
this pluralisation method will not work in languages other than english, unfortunately. It needs to be fixed in the next iteration
```suggestion
      formattedCount[severity] = `${num} ${label}${num === 1 ? "" : "s"}`;
```

Reply:
This has been applied. Resolving.